Support to skip TLS verification when image is pushed for private HTTPS registries

### Description

## Feature request

The command pushing the image build to a private registry (e.g. localhost:5000, gitea.cnoe.localtest.me:8443, etc) should allow to skip to verify the TLS certificate to avoid to get the following error 

```
[INFO] --- quarkus:3.14.4:image-push (default-cli) @ code-with-quarkus ---
[INFO] [io.quarkus.kubernetes.deployment.PropertyUtil] Kubernetes manifests are generated with 'The container port http' having default value '8080'. The app and manifests will get out of sync if the property 'quarkus.http.port' is changed at runtime.
[INFO] Testcontainers version: 1.20.1
[INFO] Loaded org.testcontainers.dockerclient.UnixSocketClientProviderStrategy from ~/.testcontainers.properties, will try it first
[INFO] Found Docker environment with Environment variables, system properties and defaults. Resolved dockerHost=unix:///var/run/docker.sock
[INFO] Docker host IP address is localhost
[INFO] Connected to docker: 
  Server Version: 5.0.3
  API Version: 1.41
  Operating System: fedora
  Total Memory: 5487 MB
[INFO] Image pull policy will be performed by: DefaultPullPolicy()
[INFO] Image name substitution will be performed by: DefaultImageNameSubstitutor (composite of 'ConfigurationFileImageNameSubstitutor' and 'PrefixingImageNameSubstitutor')
[INFO] Checking the system...
[INFO] ✔︎ Docker server version should be at least 1.6.0
[INFO] [io.quarkus.deployment.util.ExecUtil] Error response from daemon: {"errorDetail":{"message":"trying to reuse blob sha256:dd5e77a90e609b328f2e49aa60e50bd8837e505c157060c337725413ccf449f1 at destination: pinging container registry gitea.cnoe.localtest.me:8443: Get \"https://gitea.cnoe.localtest.me:8443/v2/\": tls: failed to verify certificate: x509: certificate signed by unknown authority"},"error":"trying to reuse blob sha256:dd5e77a90e609b328f2e49aa60e50bd8837e505c157060c337725413ccf449f1 at destination: pinging container registry gitea.cnoe.localtest.me:8443: Get \"https://gitea.cnoe.localtest.me:8443/v2/\": tls: failed to verify certificate: x509: certificate signed by unknown authority"}
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  2.827 s
[INFO] Finished at: 2024-09-25T08:45:58+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:3.14.4:image-push (default-cli) on project code-with-quarkus: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR]         [error]: Build step io.quarkus.container.image.docker.deployment.DockerProcessor#dockerBuildFromJar threw an exception: java.lang.RuntimeException: Execution of 'docker push gitea.cnoe.localtest.me:8443/giteaadmin/code-with-quarkus:1.0.0-SNAPSHOT' failed. See docker output for more details
[ERROR]         at io.quarkus.container.image.docker.common.deployment.CommonProcessor.containerRuntimeException(CommonProcessor.java:254)
[ERROR]         at io.quarkus.container.image.docker.common.deployment.CommonProcessor.pushImage(CommonProcessor.java:225)
```

using the following commands
```
podman machine start
export DOCKER_HOST="unix:///var/run/docker.sock"            
./mvnw install -Dquarkus.container-image.build=true \
   -Dquarkus.container-image.push=true \
   -Dquarkus.container-image.insecure=true \
   -Dquarkus.container-image.image=gitea.cnoe.localtest.me:8443/giteaadmin/code-with-quarkus:1.0.0-SNAPSHOT

or

quarkus image push \
  --registry gitea.cnoe.localtest.me:8443 \
  --group giteaadmin \
  --name code-with-quarkus \
  --tag 1.0.0-SNAPSHOT
```

Of course, that works using:
```
podman build -f src/main/docker/Dockerfile.jvm -t gitea.cnoe.localtest.me:8443/giteaadmin/code-with-quarkus:1.0.0-SNAPSHOT .
podman push gitea.cnoe.localtest.me:8443/giteaadmin/code-with-quarkus:1.0.0-SNAPSHOT --tls-verify=0
```

**Remark**: We should also offer a new parameter to the CLI to allow to consume the image URL directky without having to pass registry, group, name and tag arguments
```
quarkus image push gitea.cnoe.localtest.me:8443/giteaadmin/code-with-quarkus:1.0.0-SNAPSHOT
```

### Implementation ideas

_No response_

--- a/extensions/container-image/container-image-docker-common/deployment/src/main/java/io/quarkus/container/image/docker/common/deployment/CommonProcessor.java
+++ b/extensions/container-image/container-image-docker-common/deployment/src/main/java/io/quarkus/container/image/docker/common/deployment/CommonProcessor.java
@@ -13,7 +13,6 @@ import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
-import java.util.stream.Stream;
 
 import org.jboss.logging.Logger;
 
@@ -212,13 +211,17 @@ public abstract class CommonProcessor<C extends CommonConfig> {
                 });
     }
 
-    protected void pushImages(ContainerImageInfoBuildItem containerImageInfo, String executableName) {
-        Stream.concat(containerImageInfo.getAdditionalImageTags().stream(), Stream.of(containerImageInfo.getImage()))
-                .forEach(imageToPush -> pushImage(imageToPush, executableName));
+    protected void pushImages(ContainerImageInfoBuildItem containerImageInfo, String executableName, C config) {
+        List<String> imagesToPush = new ArrayList<>(1 + containerImageInfo.getAdditionalImageTags().size());
+        imagesToPush.add(containerImageInfo.getImage());
+        imagesToPush.addAll(containerImageInfo.getAdditionalImageTags());
+        for (String image : imagesToPush) {
+            pushImage(image, executableName, config);
+        }
     }
 
-    protected void pushImage(String image, String executableName) {
-        String[] pushArgs = { "push", image };
+    protected void pushImage(String image, String executableName, C config) {
+        String[] pushArgs = createPushArgs(image, config);
         var pushSuccessful = ExecUtil.exec(executableName, pushArgs);
 
         if (!pushSuccessful) {
@@ -226,6 +229,10 @@ public abstract class CommonProcessor<C extends CommonConfig> {
         }
 
         LOGGER.infof("Successfully pushed %s image %s", getProcessorImplementation(), image);
+    }
+
+    protected String[] createPushArgs(String image, C config) {
+        return new String[] { "push", image };
     }
 
     protected void buildImage(ContainerImageInfoBuildItem containerImageInfo,

--- a/extensions/container-image/container-image-docker/deployment/src/main/java/io/quarkus/container/image/docker/deployment/DockerProcessor.java
+++ b/extensions/container-image/container-image-docker/deployment/src/main/java/io/quarkus/container/image/docker/deployment/DockerProcessor.java
@@ -141,7 +141,7 @@ public class DockerProcessor extends CommonProcessor<DockerConfig> {
         if (!useBuildx && pushContainerImage) {
             // If not using buildx, push the images
             loginToRegistryIfNeeded(containerImageConfig, containerImageInfo, executableName);
-            pushImages(containerImageInfo, executableName);
+            pushImages(containerImageInfo, executableName, dockerConfig);
         }
 
         return containerImageInfo.getImage();

--- a/extensions/container-image/container-image-podman/deployment/src/main/java/io/quarkus/container/image/podman/deployment/PodmanConfig.java
+++ b/extensions/container-image/container-image-podman/deployment/src/main/java/io/quarkus/container/image/podman/deployment/PodmanConfig.java
@@ -7,6 +7,7 @@ import io.quarkus.container.image.docker.common.deployment.CommonConfig;
 import io.quarkus.runtime.annotations.ConfigPhase;
 import io.quarkus.runtime.annotations.ConfigRoot;
 import io.smallrye.config.ConfigMapping;
+import io.smallrye.config.WithDefault;
 
 @ConfigRoot(phase = ConfigPhase.BUILD_TIME)
 @ConfigMapping(prefix = "quarkus.podman")
@@ -16,4 +17,10 @@ public interface PodmanConfig extends CommonConfig {
      * https://docs.podman.io/en/latest/markdown/podman-build.1.html#platform-os-arch-variant
      */
     Optional<List<String>> platform();
+
+    /**
+     * Require HTTPS and verify certificates when contacting registries
+     */
+    @WithDefault("true")
+    boolean tlsVerify();
 }

--- a/extensions/container-image/container-image-podman/deployment/src/main/java/io/quarkus/container/image/podman/deployment/PodmanProcessor.java
+++ b/extensions/container-image/container-image-podman/deployment/src/main/java/io/quarkus/container/image/podman/deployment/PodmanProcessor.java
@@ -121,11 +121,16 @@ public class PodmanProcessor extends CommonProcessor<PodmanConfig> {
             if (isMultiPlatformBuild) {
                 pushManifests(containerImageInfo, executableName);
             } else {
-                pushImages(containerImageInfo, executableName);
+                pushImages(containerImageInfo, executableName, podmanConfig);
             }
         }
 
         return image;
+    }
+
+    @Override
+    protected String[] createPushArgs(String image, PodmanConfig config) {
+        return new String[] { "push", image, "push", image, String.format("--tls-verify=%b", config.tlsVerify()) };
     }
 
     private String[] getPodmanBuildArgs(String image,

